lib/pure/osproc.nim

```diff
@@ -517,18 +556,28 @@ when defined(Windows) and not defined(useNimRtl):
       si.dwFlags = STARTF_USESTDHANDLES # STARTF_USESHOWWINDOW or
       if poInteractive notin options:
         createPipeHandles(si.hStdInput, hi)
+        if hIn != INVALID_HANDLE_VALUE:
+          si.hStdInput = hIn
```

Comment:
Good catch.  They are closed before overwriting in the non-Windows code path, but not here. I will update these.